models/_sector/dex/trades/ethereum/platforms/defiswap_ethereum_base_trades.sql

```diff
@@ -0,0 +1,51 @@
+{{ config(
+    schema = 'defiswap_ethereum',
+    alias ='base_trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['tx_hash', 'evt_index']
+    )
+}}
+
+{% set project_start_date = '2020-09-09' %}
+
+WITH dexs AS
+(
+    -- defiswap
+    SELECT
+        t.evt_block_time AS block_time
+        ,t.to AS taker
+        ,t.contract_address AS maker
+        ,CAST(NULL AS DOUBLE) AS amount_usd
+        ,CASE WHEN amount0Out = '0' THEN amount1Out ELSE amount0Out END AS token_bought_amount_raw
+        ,CASE WHEN amount0In = '0' OR amount1Out = '0' THEN amount1In ELSE amount0In END AS token_sold_amount_raw
+        ,CASE WHEN amount0Out = '0' THEN f.token1 ELSE f.token0 END AS token_bought_address
+        ,CASE WHEN amount0In = '0' OR amount1Out = '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
+        ,t.contract_address AS project_contract_address
+        ,t.evt_tx_hash AS tx_hash
+        ,'' AS trace_address
```

Comment:
I am thinking `trace_address` for dex.trades should be deprecated at this point.